R/data.table.R

```diff
@@ -2887,8 +2887,13 @@ isReallyReal <- function(x) {
     ## loop continues with remainingIsub
   }
   if (length(i) == 0L) stop("Internal error in .isFastSubsettable. Please report to data.table developers")
-  ## convert i to data.table with all combinations in rows. Care is needed with names as we do
-  ## it with 'do.call' and this would cause problems if colNames were 'sorted' or 'unique'
+  ## convert i to data.table with all combinations in rows.
+  if(length(i) > 1 && prod(sapply(i, length)) > 1e4){
```

Comment:
Very minor integer discipline, but may as well always do it henceforth.

```r
if (length(i) > 1L && prod(vapply(i, length, integer(1L))) > 1e4L) {
```

Reply:
Agree on `length(i) > 1L`, but `prod` always returns numeric, so I would stick to `1e4` instead of `1e4L`. Replacing sapply with vapply is OK since it is a minor speed improvement.